src/v/cluster/log_eviction_stm.cc

```diff
@@ -142,11 +153,15 @@ ss::future<> log_eviction_stm::do_write_raft_snapshot(model::offset index_lb) {
     co_await _raft->log().stm_manager()->ensure_snapshot_exists(index_lb);
     const auto max_collectible_offset
       = _raft->log().stm_manager()->max_collectible_offset();
-    if (index_lb > max_collectible_offset) {
+    if (index_lb == max_collectible_offset) {
+        /// Cannot truncate, have already reached maximum allowable
+        co_return;
```

Comment:
I'm having a hard time reasoning about when this will fire. My suggestion was to recheck `index_lb <= _raft->last_snapshot_index())` after setting `index_lb = max_collectible_offset` at L166

Reply:
done